omnigibson/robots/behaviorbot.py

```diff
@@ -0,0 +1,677 @@
+        # If the simulator is playing, return the pose of the base_footprint link frame
+        if self._dc is not None and self._dc.is_simulating():
+            return self.base_footprint_link.get_position_orientation()
+
+        # Else, return the pose of the robot frame
+        else:
+            return super().get_position_orientation()
+
+    def set_position_orientation(self, position=None, orientation=None):
+        super().set_position_orientation(position, orientation)
+        # Move the joint frame for the world_base_joint
+        if self._world_base_fixed_joint_prim is not None:
+            if position is not None:
+                self._world_base_fixed_joint_prim.GetAttribute("physics:localPos0").Set(tuple(position))
+            if orientation is not None:
+                self._world_base_fixed_joint_prim.GetAttribute("physics:localRot0").Set(Gf.Quatf(*np.float_(orientation)[[3, 0, 1, 2]]))
+
+    @property
+    def assisted_grasp_start_points(self):
+        side_coefficients = {"lh": np.array([1, -1, 1]), "rh": np.array([1, 1, 1])}
+        return {
+            arm: [
+                GraspingPoint(link_name=f"{arm}_{PALM_LINK_NAME}", position=PALM_BASE_POS),
+                GraspingPoint(link_name=f"{arm}_{PALM_LINK_NAME}", position=PALM_CENTER_POS * side_coefficients[arm]),
+                GraspingPoint(
+                    link_name=f"{arm}_{THUMB_LINK_NAME}", position=THUMB_1_POS * side_coefficients[arm]
+                ),
+                GraspingPoint(
+                    link_name=f"{arm}_{THUMB_LINK_NAME}", position=THUMB_2_POS * side_coefficients[arm]
+                ),
+            ]
+            for arm in self.arm_names
+        }
+
+    @property
+    def assisted_grasp_end_points(self):
+        side_coefficients = {"lh": np.array([1, -1, 1]), "rh": np.array([1, 1, 1])}
+        return {
+            arm: [
+                GraspingPoint(link_name=f"{arm}_{finger}", position=FINGER_TIP_POS * side_coefficients[arm])
+                for finger in FINGER_TIP_LINK_NAMES
+            ]
+            for arm in self.arm_names
+        }
+
+    def update_hand_contact_info(self):
+        """
+        Helper function that updates the contact info for the hands and body. 
+        Can be used in the future with device haptics to provide collision feedback.
+        """
+        self.part_is_in_contact["body"] = len(self.links["body"].contact_list())
+        for hand_name in self.arm_names:
+            self.part_is_in_contact[hand_name] = len(self.eef_links[hand_name].contact_list()) \
+               or np.any([len(finger.contact_list()) for finger in self.finger_links[hand_name]])
+    
+    def deploy_control(self, control, control_type, indices=None, normalized=False):
+        """
+        Overwrites controllable_object.deploy_control to make arm revolute joints set_target=False
+        This is needed because otherwise the hand will rotate a full circle backwards when it overshoots pi
```

Comment:
i wonder if this is still necessary on 2023.1.0

Reply:
I tried it yesterday and it's still necessary unfortunately